File: TUF/String/FindPositionOfSubString.java
import java.util.Scanner;

public class FindPositionOfSubString {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String str = input.next();
        System.out.print("Enter the sub string: ");
        String subStr = input.next();
        input.close();
        System.out.println(findSubStringPosition(str, subStr));
    }

    private static int findSubStringPosition(String str, String subStr) {
        int strLen = str.length();
        int subStrLen = subStr.length();
        if(subStrLen > strLen) {
            throw new Error("Please enter a valid sub string..");
        }else {
            for(int i = 0; i<strLen; i++) {
                int temp = i;
                int j = 0;
                for(j = 0; j<subStrLen; j++) {
                    if(str.charAt(temp) != subStr.charAt(j)){
                        break;
                    }
                    temp++;
                }
                if (j == subStrLen) {
                    return i;
                }
            }
            return -1;
        }
    }
}

File: TUF/String/FindPositionOfSubString.ts
function findSubStringPosition(str: string, subStr: string): number {
    let strLen: number = str.length;
    let subStrLen: number = subStr.length;
    if (strLen < subStrLen) {
        throw new Error("Please enter a valid sub string...");
    } else {
        for (let i: number = 0; i < strLen; i++) {
            let temp: number = i;
            let j: number = 0;
            for (j = 0; j < subStrLen; j++) {
                if (str.charAt(temp) !== subStr.charAt(j)) {
                    break;
                }
                temp++;
            }
            if (j == subStrLen) {
                return i;
            }
        }
        return -1;
    }
}

let str: string = "takeuforward";
let subStr: string = "forward";
console.log(findSubStringPosition(str, subStr));

export { };